pbft/pbft.cpp

```diff
@@ -80,6 +82,18 @@ pbft::start()
                 this->audit_heartbeat_timer->expires_from_now(heartbeat_interval);
                 this->audit_heartbeat_timer->async_wait(
                         std::bind(&pbft::handle_audit_heartbeat_timeout, shared_from_this(), std::placeholders::_1));
+
+                auto weak_this = this->weak_from_this();
+                this->failure_detector->register_failure_handler(
```

Comment:
With c++17 you can avoid the copy...

[auto weak_this=this->weak_from_this()]()
{
...

Reply:
Sorry, how? I'm not following.

Reply:
oops... it stripped my formatting for some reason:

```
[auto weak_this = this->weak_from_this()]()
{

}
```

Reply:
You sure? I get 
```/home/isabel/swarmDB/pbft/pbft.cpp: In lambda function:
/home/isabel/swarmDB/pbft/pbft.cpp:87:26: error: expected identifier before ‘auto’
                         [auto weak_this = this->weak_from_this()]()
```
GCC 8.0.1

Reply:
sorry auto isn't allowed. My mistake. But I would put it there anyway.

Reply:
It doesn't like that either
```
/home/isabel/swarmDB/pbft/pbft.cpp: In lambda function:
/home/isabel/swarmDB/pbft/pbft.cpp:88:26: error: capture of non-variable ‘std’ 
                         [std::weak_ptr<pbft> weak_this = this->weak_from_this()]()
                          ^~~
<built-in>: note: ‘namespace std { }’ declared here
/home/isabel/swarmDB/pbft/pbft.cpp:88:29: error: expected ‘,’ before ‘::’ token
                         [std::weak_ptr<pbft> weak_this = this->weak_from_this()]()
                             ^~
```

Reply:
gah... my brain is turning to mush... drop the type and use like:

```
[weak_this = this->weak_from_this()]()
{
}
```